client/src/main/scala/org/apache/celeborn/client/WorkerStatusTracker.scala

```diff
@@ -142,32 +146,34 @@ class WorkerStatusTracker(
                 StatusCode.PUSH_DATA_CONNECTION_EXCEPTION_SLAVE |
                 StatusCode.PUSH_DATA_TIMEOUT_MASTER |
                 StatusCode.PUSH_DATA_TIMEOUT_SLAVE
-                if current - registerTime < workerExcludedExpireTimeout =>
-              true
-            case StatusCode.UNKNOWN_WORKER => true
-            case _ => false
+                if current - registerTime < workerExcludedExpireTimeout => // reserve
+            case _ => blacklist.remove(workerInfo)
           }
-        }.asJava
-      val reservedBlackList = new ShuffleFailedWorkers()
-      reservedBlackList.putAll(reserved)
-      blacklist.clear()
-      blacklist.putAll(
-        res.blacklist.asScala.map(_ -> (StatusCode.WORKER_IN_BLACKLIST -> current)).toMap.asJava)
-      blacklist.putAll(
-        res.unknownWorkers.asScala.map(_ -> (StatusCode.UNKNOWN_WORKER -> current)).toMap.asJava)
-      // put reserved blacklist at last to cover blacklist's local status.
-      blacklist.putAll(reservedBlackList)
-
-      val workerStatus = new WorkersStatus(res.unknownWorkers, newShutdownWorkers)
-      workerStatusListeners.asScala.foreach {
-        listener =>
+      }
+
+      if (!res.blacklist.isEmpty) {
+        blacklist.putAll(res.blacklist.asScala.filterNot(blacklist.containsKey).map(
+          _ -> (StatusCode.WORKER_IN_BLACKLIST -> current)).toMap.asJava)
```

Comment:
```
.map(_ -> (StatusCode.WORKER_IN_BLACKLIST -> current)).toMap.asJava)
```

In this kind of place, a transformation can be used in one line, it is better not to separate it